lib/Db/MessageMapper.php

```diff
@@ -176,10 +176,18 @@ public function findUidsForIds(Mailbox $mailbox, array $ids) {
 			->from($this->getTableName())
 			->where(
 				$query->expr()->eq('mailbox_id', $query->createNamedParameter($mailbox->getId(), IQueryBuilder::PARAM_INT), IQueryBuilder::PARAM_INT),
-				$query->expr()->in('id', $query->createNamedParameter($ids, IQueryBuilder::PARAM_INT_ARRAY), IQueryBuilder::PARAM_INT_ARRAY)
+				$query->expr()->in('id', $query->createParameter('ids'), IQueryBuilder::PARAM_INT_ARRAY)
 			);
 
-		return $this->findUids($query);
+		$chunks = array_chunk($ids, 1000);
+
+		$results = [];
+		foreach ($chunks as $chunk) {
```

Comment:
y u no `array_map`?

Reply:
I think this is more readable, but I can do

```php
		return array_merge(... array_map(function(array $ids) use ($query) {
			$query->setParameter('ids', $ids, IQueryBuilder::PARAM_INT_ARRAY);
			return $this->findUids($query);
		}));
```

if you want?

Reply:
I'm fine with either